yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -464,7 +472,7 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments,
   /**
    * Actor that communicates with the driver in client deploy mode.
```

Comment:
I don't understand. In both client and cluster modes, this actor is used to communicate with the driver. It's just that in cluster mode since the driver and the AM are run in the same JVM, the communication is local.